src/Microsoft.CodeAnalysis.Razor/DefaultTagHelperDescriptorFactory.cs

```diff
@@ -894,323 +495,299 @@ internal static string ToHtmlCase(string name)
 
-                var value = ParseCssValue(errorSink);
-
-                if (value == null)
+                private bool EnsureNotAtEnd(RequiredAttributeDescriptorBuilder builder)
                 {
-                    // Couldn't parse value
-                    return null;
-                }
+                    if (AtEnd)
+                    {
+                        var diagnostic = RazorDiagnosticFactory.CreateTagHelper_CouldNotFindMatchingEndBrace(_requiredAttributes);
+                        builder.AddDiagnostic(diagnostic);
 
-                PassOptionalWhitespace();
+                        return false;
+                    }
 
-                if (At(']'))
-                {
-                    // Move past the ending bracket.
-                    _index++;
-                }
-                else if (AtEnd)
-                {
-                    errorSink.OnError(
-                        SourceLocation.Zero,
-                        Resources.FormatTagHelperDescriptorFactory_CouldNotFindMatchingEndBrace(_requiredAttributes),
-                        length: 0);
-                    return null;
-                }
-                else
-                {
-                    errorSink.OnError(
-                        SourceLocation.Zero,
-                        Resources.FormatTagHelperDescriptorFactory_InvalidRequiredAttributeCharacter(Current, _requiredAttributes),
-                        length: 0);
-                    return null;
+                    return true;
                 }
 
-                return new TagHelperRequiredAttributeDescriptor
-                {
-                    Name = attributeName,
-                    NameComparison = TagHelperRequiredAttributeNameComparison.FullMatch,
-                    Value = value,
-                    ValueComparison = valueComparison.Value,
-                };
-            }
-
-            private bool EnsureNotAtEnd(ErrorSink errorSink)
-            {
-                if (AtEnd)
+                private bool At(char c)
                 {
-                    errorSink.OnError(
-                        SourceLocation.Zero,
-                        Resources.FormatTagHelperDescriptorFactory_CouldNotFindMatchingEndBrace(_requiredAttributes),
-                        length: 0);
-
-                    return false;
+                    return !AtEnd && Current == c;
                 }
 
-                return true;
-            }
-
-            private bool At(char c)
-            {
-                return !AtEnd && Current == c;
-            }
-
-            private void PassOptionalWhitespace()
-            {
-                while (!AtEnd && (Current == ' ' || Current == '\t'))
+                private void PassOptionalWhitespace()
                 {
-                    _index++;
+                    while (!AtEnd && (Current == ' ' || Current == '\t'))
+                    {
+                        _index++;
+                    }
                 }
```

Comment:
It's really nice overall to see the changes to this class, this is much cleaner than before.